modules/apps/foundation/portal-configuration/portal-configuration-settings/src/main/java/com/liferay/portal/configuration/settings/internal/SettingsLocatorHelperImpl.java

```diff
@@ -68,6 +69,29 @@
 @DoPrivileged
 public class SettingsLocatorHelperImpl implements SettingsLocatorHelper {
 
+	@Override
+	public Settings getCompanyConfigurationBeanSettings(
+		long companyId, String configurationPid, Settings parentSettings) {
+
+		if (!_configurationBeanClasses.containsKey(configurationPid)) {
+			return parentSettings;
+		}
+
+		ScopeKey scopeKey = new ScopeKey(
+			_configurationBeanClasses.get(configurationPid),
+			ExtendedObjectClassDefinition.Scope.COMPANY,
+			String.valueOf(companyId));
+
+		if (!_scopedConfigurationBeanProvider.has(scopeKey)) {
+			return parentSettings;
+		}
+
+		return new ConfigurationBeanSettings(
+			_configurationBeanLocationVariableResolvers.get(
+				scopeKey.getObjectClass()),
+			_scopedConfigurationBeanProvider.get(scopeKey), parentSettings);
+	}
+
```

Comment:
Maybe we can create a helper method here to create the ConfigurationBeanSettings to simplify the code. There's a lot of similar code here.

```
_getScopedConfigurationBeanSettings(
String scopePrimKey, ExtendedObjectClassDefinition.Scope scope, String configurationPid,
Settings parentSettings)
```

Reply:
Oops, that was actually part of the original pull from Stian that I forgot to do.  Because, basically, they're all identical methods except for which scope they use.  Thanks for the catch 👍